client/data/agency-dashboard/use-sync-jetpack-connection-status.ts

```diff
@@ -0,0 +1,61 @@
+import { useQuery, useQueryClient } from '@tanstack/react-query';
+import { useContext, useEffect, useMemo } from 'react';
+import SitesOverviewContext from 'calypso/jetpack-cloud/sections/agency-dashboard/sites-overview/context';
+import { Site } from 'calypso/jetpack-cloud/sections/agency-dashboard/sites-overview/types';
+import wpcom from 'calypso/lib/wp';
+
+const useSyncJetpackConnectionStatus = (
```

Comment:
FWIW this hook seems like something that should likely live in another, more generic place, because it's not specifically about the agency dashboard; it's more about querying the jetpack connection status on any Jetpack site, like we're doing in `requestConnectionStatus` [here](https://github.com/Automattic/wp-calypso/blob/c625128324f8435185e34fab212443226e3f6ee1/client/state/site-connection/actions.js#L17).

Reply:
I tend to agree, but only for the query part. As we're not only querying but also syncing I think its name and location are fitting. We could, however, extract the query part and make it a generic hook stored in `client/data/jetpack` and use that one here.